source/_posts/2018-01-09-automatically-managing-personal-and-work-git-configurations.markdown

```diff
@@ -0,0 +1,87 @@
+---
+layout: post
+title: "Automatically managing personal and work git configurations"
+category:
+comments: true
+author:
+  name: Federico Kauffman
+  email: federico@wyeworks.com
+  twitter_handle: fedekauffman
+  github_handle: fedekau
+  image: /images/team/federico-kauffman.jpg
+  description: Software Engineer at WyeWorks. Currently working with Javascript and Ruby. Learnaholic.
+---
+
+There was a time where I found myself switching between my personal computer and working computer, and that was really annoying, after a time switching back and forth I decided to just use one computer.
+
+After choosing the single computer approach, what irritated me the most was having to remember to switch my git user, email and keys when moving from one work repository to a personal repository and vice versa. I would mostly forget to do that and that caused that every now and then I would find commits at work done with my personal email or the other way around.
+
+At work most of the repositories are private, but at home most of my work is public, which means that my work email would be public, I don't like that. So in this post I will share one way you can forget about having to switch and let the computer work for you and do it automatically.
+
+<!--more-->
+
+## Step 1: Configure your SSH keys
+
+If you are using SSH to interact with you Git server (probably GitHub) read on, otherwise skip this step. Or even better: **Start using SSH**.
+
+Let's say you have two ssh key-pairs, one for `user@work.com` and one for `user@personal.com`. You want that when pushing, cloning, etc. from a personal or a work repository you use your corresponding ssh key-pair. That's easy, modify or create a `~/.ssh/config` file to look similar to this:
+
+{% codeblock lang:bash %}
+#personal
+Host github.com-personal
+  HostName github.com
+  User personal
+  IdentityFile ~/.ssh/id_rsa_personal
+  PreferredAuthentications publickey
+
+#work
+Host github.com-work
+  HostName github.com
+  User work
+  IdentityFile ~/.ssh/id_rsa_work
+  PreferredAuthentications publickey
+{% endcodeblock %}
+
+Replace `personal` with your personal git user and `work` with your user from work and this step is done.
+
+## Step 2: Add user details to work and personal folders
+
+In this step I am assuming you have all your work stuff under a `~/work` folder and all your personal stuff under a `~/personal` folder. If you don't, then you either have to move things on your computer or adapt this step to your needs.
+
+In each folder add one file called `.gitconfig`, you should have one file `~/work/.gitconfig` and one `~/personal/.gitconfig`. In each file you should fill in your information accordingly, for example:
+
+{% codeblock lang:bash %}
+[user]
+  email = user@work.com
+  name = user
```

Comment:
I would use a name different from `user`, because there is a value `user` in the SSH config file that is not the same. Maybe `John Doe`?